lib/internal/warn.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+// A process Warning is similar in structure to an Error except that they
+// are emitted by the process.on('warn') event rather than being thrown.
+
+const _message = Symbol('message');
+const _name = Symbol('name');
+const prefix = `(${process.release.name}:${process.pid}) `;
+
+function warning(name, message, constructor) {
+  const err = new Error(`${prefix}${message}`);
+  err.name = name;
+  Error.captureStackTrace(err, constructor || warning);
+  return err;
+}
+
+function emit(warning) {
+  process.nextTick(() => process.emit('warn', warning));
+}
+
+// each of the following use the same two-argument signature:
+//  * message is the warning message. this is handled as a literal. No
+//    formatting is done
+//  * warned is a truthy or falsy value. It is used as an indication
+//    if the warning has already been issued or not... e.g.
+//
+//    var warned;
+//    function doWarning() {
+//      warned = Warning.securityWarning('do not do that', warned);
+//    }
+//    doWarning(); doWarning();
+//    // the warning is only emitted once.
+}
+
+warning.deprecationWarning = function(message, warned) {
+  if (process.noDeprecation || warned) return true;
+  const w = warning('DeprecationWarning', message, warning.deprecationWarning);
+  // handle --throw-deprecation appropriately
+  if (process.throwDeprecation)
+    throw warning;
+  else
+    emit(warning);
+  return true;
```

Comment:
Are the return values used?  The only place I could find is `util._printDeprecationMessage()` and there you could just `return warned`.  If you drop the return values here, you can condense the code a little more.